drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -144,92 +222,202 @@ local function assign_child_profile(device, child_ep)
+end
+
 local function initialize_switch(driver, device)
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
-  table.sort(switch_eps)
-  -- Since we do not support bindings at the moment, we only want to count On/Off
-  -- clusters that have been implemented as server. This can be removed when we have
+  local button_eps = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.Feature.MOMENTARY_SWITCH})
+  local all_eps = {}
+
+  local profile_name = nil
+
+  local component_map = {}
+  local current_component_number = 2
+  local component_map_used = false
+  local parent_child_device = false
+
+  if #switch_eps == 0 and #button_eps == 0 then
+    return
+  end
+
+  for _,ep in ipairs(switch_eps) do
+    table.insert(all_eps, ep)
+  end
+  for _,ep in ipairs(button_eps) do
+    table.insert(all_eps, ep)
+  end
+  table.sort(all_eps)
+  -- Since we do not support bindings at the moment, we only want to count clusters
+  -- that have been implemented as server. This can be removed when we have
   -- support for bindings.
-  local num_server_eps = 0
+  local num_switch_server_eps = 0
   local main_endpoint = find_default_endpoint(device)
-  for _, ep in ipairs(switch_eps) do
-    if device:supports_server_cluster(clusters.OnOff.ID, ep) then
-      num_server_eps = num_server_eps + 1
+
+  for _, ep in ipairs(all_eps) do
+    if device:supports_server_cluster(clusters.Switch.ID, ep) then
+      -- Configure MCD for button endpoints
+      if tbl_contains(STATIC_BUTTON_PROFILE_SUPPORTED, #button_eps) then
+        if ep ~= main_endpoint then
+          component_map[string.format("button%d", current_component_number)] = ep
+          current_component_number = current_component_number + 1
+        else
+          component_map["main"] = ep
+        end
+        component_map_used = true
+      end
+    elseif device:supports_server_cluster(clusters.OnOff.ID, ep) then
+      -- Create child devices for non-main switch endpoints
+      num_switch_server_eps = num_switch_server_eps + 1
       if ep ~= main_endpoint then -- don't create a child device that maps to the main endpoint
-        local name = string.format("%s %d", device.label, num_server_eps)
+        local name = string.format("%s %d", device.label, num_switch_server_eps)
         local child_profile = assign_child_profile(device, ep)
         driver:try_create_device(
-          {
-            type = "EDGE_CHILD",
-            label = name,
-            profile = child_profile,
-            parent_device_id = device.id,
-            parent_assigned_child_key = string.format("%d", ep),
-            vendor_provided_label = name
-          }
+            {
+              type = "EDGE_CHILD",
+              label = name,
+              profile = child_profile,
+              parent_device_id = device.id,
+              parent_assigned_child_key = string.format("%d", ep),
+              vendor_provided_label = name
+            }
```

Comment:
nit: remove added whitespace?

Reply:
Fixed in 56f40cf534af547dee93afcff6d524d82419be86